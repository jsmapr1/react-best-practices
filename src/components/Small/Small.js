import React from 'react';
import './rating.css';
import Ansel from './ansel.jpg';

function getRating(count) {
  return Array(count).fill('').map(item => {
    return {
      rating: Math.ceil(Math.random() * 4),
      reviewer: ['Mike', 'Joe', 'Bill', 'Evan', 'Brian', 'Frank'][Math.floor(Math.random() * 6)]
    }
  })
}

const defaultPhoto = {
  src: Ansel,
  description: {
    name: 'Tetons and Snake River',
    photographer: 'Ansel Adams',
    location: 'Tetons'
  },
  ratings: getRating(10)
}

export default function Small({ photo = defaultPhoto}) {
  return(
    <div>
      <img src={photo.src}/>
      <div className="Description">
        <ul>
          {Object.keys(photo.description).map(key => {
            return (<li key={key}>{key}: {photo.description[key]}</li>)
          })}
        </ul>
      </div>
      <div>
        {photo.ratings.map(rating => {
          return (
            <div className="rating">
              <h3>Rating</h3>
              {Array(rating.rating).fill(<span />)}
              <div>Reviewer: {rating.reviewer}</div>
            </div>
          )
        })}
      </div>
    </div>
  )
}
