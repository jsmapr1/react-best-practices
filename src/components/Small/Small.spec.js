import React from 'react';
import { shallow } from 'enzyme';

import Small from './Small';

describe('Small', () => {
  it('should render Small', () => {
    const wrapper = shallow(
      <Small />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
