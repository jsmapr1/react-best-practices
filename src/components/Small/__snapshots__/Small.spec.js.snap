// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small should render Small 1`] = `
<div>
  <img
    src="ansel.jpg"
  />
  <div
    className="Description"
  >
    <ul>
      <li
        key="name"
      >
        name
        : 
        Tetons and Snake River
      </li>
      <li
        key="photographer"
      >
        photographer
        : 
        Ansel Adams
      </li>
      <li
        key="location"
      >
        location
        : 
        Tetons
      </li>
    </ul>
  </div>
  <div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <div>
        Reviewer: 
        Brian
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <div>
        Reviewer: 
        Joe
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <div>
        Reviewer: 
        Brian
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <span />
      <div>
        Reviewer: 
        Frank
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <div>
        Reviewer: 
        Brian
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <span />
      <span />
      <div>
        Reviewer: 
        Brian
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <span />
      <div>
        Reviewer: 
        Mike
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <div>
        Reviewer: 
        Brian
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <span />
      <span />
      <div>
        Reviewer: 
        Bill
      </div>
    </div>
    <div
      className="rating"
    >
      <h3>
        Rating
      </h3>
      <span />
      <span />
      <span />
      <div>
        Reviewer: 
        Bill
      </div>
    </div>
  </div>
</div>
`;
